### PR TITLE
add zeromq library / cppzmq bindings

### PR DESCRIPTION
**Pull request by MoKob**
Libraries to allow for communication between tools via protobuf and sockets.

### CODE REVIEW

**Comment by MoKob** [comment]
@springmeyer could you have a look here? I would love to use these libs for my current project and sadly cppzmq isn't bundled in brew / libzmq is pretty outdated on ubuntu.

**Comment by springmeyer** [comment]
@MoKob this looks great. I made a few minor adjustments (with details in the commit message). I think this is ready to go. The steps you can take next are:

- Set up your `MASON_TRAVIS_TOKEN` by following https://github.com/mapbox/mason#trigger
- Publish builds from this branch by running:
```
./mason trigger cppzmq 4.2.1
./mason trigger libzmq 4.2.2
```
- Once those builds are green on travis they will have published (libzmq for both linux and osx and cppzmq headers via linux job).
- Then you can merge this branch and add to the changelog in `master.
- If you need this in a release feel free to roll a release per https://github.com/mapbox/mason/blob/master/contributing.md#release-process

File: CHANGELOG.md [hunk 1]
```diff
@@ -96,4 +101,4 @@ Changes: https://github.com/mapbox/mason/compare/v0.1.1...v0.2.0
  - Improved llvm >= 3.8.1 packages to support `-flto` on linux
 
 ## 0.1.0
- - First versioned release
\ No newline at end of file
+ - First versioned release
```
Comment (brunoabinader) [inline CHANGELOG.md]: Carriage return mismatch?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Mason releases
 
+## 0.8.0
+ - Added
+    - zeromq 4.2.2
+    - cppzmq 4.2.1
+
```
Comment (kkaefer) [inline CHANGELOG.md]: We don't generally add packages to the CHANGELOG.

File: CHANGELOG.md (hunk 1 above)
Comment (MoKob) [inline CHANGELOG.md]: No idea what happens there. Newlines?

Comment (brunoabinader) [inline CHANGELOG.md]: Seem like the text editor used is adding them on purpose. In this case, you can e.g. use `git add -p` and avoid that particular change.

File: CHANGELOG.md (hunk 2 above)
Comment (MoKob) [inline CHANGELOG.md]: From conversation in chat: packages have been added in recent release notes. If there are no objections, I would continue with this progress of adding them.

**Comment by MoKob** [comment]
@springmeyer can you have a final look and click the green button if you are OK with it?

**Comment by springmeyer** [comment]
@MoKob yes, will do today.

File: CHANGELOG.md (hunk 2 above)
Comment (springmeyer) [inline CHANGELOG.md]: 👍  Yes, we now and always add packages to the CHANGELOG.

**Comment by springmeyer** [comment]
rebased (to handle CHANGELOG conflict) and merged.

**Pull request merged**